.circleci/config.yml

```
@@ -0,0 +1,50 @@
+---
+x-defs:
+  default_docker: &rust_docker [{image: "circleci/rust:latest"}]
+
+version: 2
+jobs:
+  fetch-deps:
+    docker: *rust_docker
+    steps:
+      - run: rustup default nightly
+      - checkout
+      - run: cargo fetch
+
+  test:
+    docker: *rust_docker
+    steps:
+      - run: rustup default nightly
+      - checkout
+      - run: cargo test
+
+  rustfmt:
+    docker: *rust_docker
+    steps:
+      - run: rustup default nightly
+      - checkout
+      - run: rustup component add rustfmt-preview --toolchain nightly
+      - run: cargo fmt --all -- --check
+
+  compile:
+    docker: *rust_docker
+    steps:
+      - run: rustup default nightly
+      - checkout
+      - run: cargo build -j8 --release
```

Don't we only have 2 cores on Circle?

```suggestion
      - run: cargo build -j2 --release
```